### DISCUSSION TITLE
Issue with PTFs needed for debug function to work

### DISCUSSION THREAD

**Discussion by blhowie66** [codefori/vscode-ibmi#1919] [Q&A]
Hello.  I was attempting to get the debug function working for my version of Code for IBM i.  I have the latest version, 2.8.1.  I noticed a discrepancy between the list of needed PTFs for 7.4.  On the IBM i Debug welcome screen it says that SI80364 and SI81031 are needed for 7.4.  Those are now installed.   When I go into a particular source member and try to debug using the icon in the top right of the window, it walks me through the setting up of the debug service, and acts like I don't have the debug PTFs installed, and says that I need **SI83683** and SI81031.  So the SI83683 is now installed, and when it tries to do the "Setup Certificates" portion of setting up the debug service, it tells me the "Debug PTF is not installed", but it is.  Any thoughts on this?

**Comment by mlstoppa** [comment]
I do have the same problem. Named PTFs are installed on a V7R4 system, but Code for I doesn't recognize them.

**Comment by sebjulliand** [comment]
@blhowie66 , @mlstoppa ,
to tell if the debug PTF is installed, Code for i will look for the script `startDebugService.sh` under `/QIBM/ProdData/IBMiDebugService/bin`.

First, make sure this file exists and it is accessible to you.
Next, since Code for i only tests this once and caches the information, you'll need to disconnect and then reload the server settings. Right click on your connection and run this:
![image](https://github.com/codefori/vscode-ibmi/assets/11096890/9a35ef6a-d0ea-41fa-b3c5-bbbadda6eff0)

At this point, Code for i should have changed its mind about the debugger being installed. 😊

**Comment by mlstoppa** [comment]
Hi Seb, 
that's great to know.
